### PR TITLE
Feature/mdr 315

### PR DESCRIPTION
**Pull request by costezki**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -28,43 +28,19 @@ stop-fuseki:
...
 	@ docker-compose --file docker/docker-compose.yml --env-file docker/.env down
 
-fuseki-create-test-dbs: | start-fuseki
-	@ echo "$(BUILD_PRINT)Building dummy 'dev' dataset at http://localhost:$(if $(RDF_FINGERPRINTER_FUSEKI_PORT),$(RDF_FINGERPRINTER_FUSEKI_PORT),unknown port)/$$/datasets"
-	@ sleep 5
-	@ curl --anyauth --user 'admin:$(RDF_FINGERPRINTER_FUSEKI_ADMIN_PASSWORD)' -d 'dbType=mem&dbName=dev'  'http://localhost:$(RDF_FINGERPRINTER_FUSEKI_PORT)/$$/datasets'
+fuseki-create-test-dbs: start-fuseki
+	@ echo "$(BUILD_PRINT) Uploading the datasets"
+	@ sleep 3
+	@ curl -X DELETE --anyauth --user 'admin:admin' http://localhost:$(RDF_FINGERPRINTER_FUSEKI_PORT)/dev/data
```
Comment (cosletmihai) [inline Makefile:34]: will the password will always be `admin`?

File: fingerprint_report_templates/fingerprint_report/templates/macros.html [hunk 2]
```diff
@@ -58,4 +58,129 @@
...
         {{ render_error("How did you get here? did you forget to use 'render_fetch_results' macro?") }}
     {% endif %}
+{%- endmacro %}
+
+{############################################}
+{# fingerpritinging macros #}
+{############################################}
+
+{# select all the graphs is the default endpoint #}
+{% macro select_graphs() -%}
```
Comment (cosletmihai) [inline fingerprint_report_templates/fingerprint_report/templates/macros.html:68]: `select_all_graphs` seems a more explicit name for this method.

File: fingerprint_report_templates/fingerprint_report/templates/macros.html [hunk 3]
```diff
@@ -58,4 +58,129 @@
...
+        {% if not content.empty %}
+            {% for row in content.itertuples() %}
+                {{ caller_(row.graph) }}
+            {% endfor %}
+        {% endif %}
+    {% endcall %}
+{%- endmacro %}
+
+{# select all the classes is the given graph #}
+{% macro select_classes_in_graph(graph=None) -%}
```
Comment (cosletmihai) [inline fingerprint_report_templates/fingerprint_report/templates/macros.html:87]: `select_distinct_class_types` (I'd not include `graph` in the name as it isn't a required parameter, so calling 
```python 
select_distinct_class_types_in_graph()
``` 
without any parameters will seem like an error

File: fingerprint_report_templates/fingerprint_report/templates/macros.html [hunk 4]
```diff
@@ -58,4 +58,129 @@
...
+{############################################}
+
+{# select all the graphs is the default endpoint #}
+{% macro select_graphs() -%}
+    {% set select_query =
+    """select distinct ?graph
+    {graph ?graph {?s ?p []}}""" %}
+    {% set content, error = from_endpoint(conf.default_endpoint).with_query(select_query).fetch_tabular() %}
+    {# caller_ is a workaround necessary to handle two levels of callers #}
+    {% set caller_ = caller %}
```
Comment (cosletmihai) [inline fingerprint_report_templates/fingerprint_report/templates/macros.html:74]: neat. Does this allow only for 2 levels of context managers?

File: fingerprint_report_templates/fingerprint_report/templates/macros.html [hunk 5]
```diff
@@ -58,4 +58,129 @@
...
+            {% for row in content.itertuples() %}
+                {{ caller_(row.cls) }}
+            {% endfor %}
+        {% endif %}
+    {% endcall %}
+{%- endmacro %}
+
+
+{# fingerprint a given class in a given graph #}
+{% macro fingerprint_class_in_graph(class, graph=None) -%}
```
Comment (cosletmihai) [inline fingerprint_report_templates/fingerprint_report/templates/macros.html:105]: same remark about `in_graph`

File: fingerprint_report_templates/fingerprint_report/templates/macros.html [hunk 6]
```diff
@@ -58,4 +58,129 @@
...
+        (GROUP_CONCAT(distinct ?tp; separator=', ') as ?property_type)
+        (count(distinct ?s) as ?subject_cnt)
+        (count(distinct ?o) as ?object_cnt)
+        (count(*) as ?pattern_cnt)
+        (if(?subject_cnt < ?total_distinct_s, 0, min(?sp_star)) as ?min)
+        (max(?sp_star) as ?max)
+        (avg(?sp_star) * ?subject_cnt/?total_distinct_s as ?avg)
+    "~from_statement~"
+    WHERE {
+        values ?ts {<"~class+">}
```
Comment (cosletmihai) [inline fingerprint_report_templates/fingerprint_report/templates/macros.html:119]: typo: `~`

File: fingerprint_report_templates/fingerprint_report/templates/macros.html [hunk 7]
```diff
@@ -24,19 +24,19 @@
     {% endif %}
 {%- endmacro %}
 
-
+{# draw a generic pandas table#}
 {% macro pandas_table(df, caption, column_labels={}) -%}
```
Comment (cosletmihai) [inline fingerprint_report_templates/fingerprint_report/templates/macros.html:28]: add the verb to the method: `render_pandas_table` to be more explicit about what the macro does

File: fingerprint_report_templates/fingerprint_report/templates/macros.html [hunk 8]
```diff
@@ -58,4 +58,129 @@
...
+
+{# build fingerpring for each class in every graph from the endpoint #}
+{% macro build_default_endpoint_fingerprint(inverted_prefixes=None) -%}
+    {% call(graph) select_graphs() %}
+        <h2>{{ 'Named graph ' + graph|safe + " fingerprints" if graph else "Default graph fingerprints" }}</h2>
+        <section>
+            {% call(class) select_classes_in_graph(graph) %}
+                {% call(fingerprint_content) fingerprint_class_in_graph(class) %}
+                    {{ render_class_fingerprint(fingerprint_content, class, graph, inverted_prefixes) }}
+                {% endcall %}
```
Comment (cosletmihai) [inline fingerprint_report_templates/fingerprint_report/templates/macros.html:182]: hardcore call.

this `call(something)` acts as a `yield`?

**Review by cosletmihai — APPROVED** [review]

File: fingerprint_report_templates/fingerprint_report/templates/macros.html (hunk 8 above)
Comment (costezki) [reply fingerprint_report_templates/fingerprint_report/templates/macros.html:197]: behind the scene jinja2 most likely uses generators

**Pull request merged**